core/src/main/java/io/grpc/transport/Http2ClientStream.java

```diff
@@ -136,8 +136,7 @@ protected void transportDataReceived(Buffer frame, boolean endOfStream) {
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError = Status.INTERNAL.withDescription("Recevied EOS on DATA frame");
-        frame.close();
-        inboundTransportError(transportError);
+        inboundTransportError(transportError, true);
```

Comment:
MessageDeframer should have been stalling sooner than it was. Created https://github.com/grpc/grpc-java/pull/191